src/chocolatey/infrastructure/registration/AssemblyResolution.cs

```diff
@@ -31,6 +31,18 @@ public class AssemblyResolution
 
         private static readonly ConcurrentDictionary<string, IAssembly> _loadedAssemblies = new ConcurrentDictionary<string, IAssembly>();
 
+        /// <summary>
+        /// Resolves or loads an assembly. If an assembly is already loaded, no need to reload it.
+        /// </summary>
+        /// <param name="assemblySimpleName">Simple Name of the assembly, such as "chocolatey"</param>
+        /// <param name="publicKeyToken">The public key token.</param>
+        /// <param name="assemblyFileLocation">The assembly file location. Typically the path to the DLL on disk.</param>
+        /// <returns>An assembly</returns>
+        /// <exception cref="Exception">Unable to enter synchronized code to determine assembly loading</exception>
+        public static IAssembly resolve_or_load_assembly(string assemblySimpleName, string publicKeyToken, string assemblyFileLocation) {

```

Comment:
If the intention is to remove this, should we not mark it as obsolete?

Reply:
From what Cory told be on Slack, it is being used by Chocolatey GUI, and since it is just an overload of the other method there isn't really a need to deprecate it.

Doesn't hurt to just keep it around.

Reply:
I'm good with marking this as obsolete if we want. A benefit of not obsoleting it is that anyone currently using Chocolatey.Lib doesn't need to do anything since this just calls the new method with the default value of that method anyways.